src/library/librarycontrol.cpp

```diff
@@ -1059,30 +1076,23 @@ void LibraryControl::slotDecrementFontSize(double v) {
     }
 }
 
-void LibraryControl::slotTrackColorPrev(double v) {
-    if (!m_pLibraryWidget) {
+void LibraryControl::slotTrackColorSelector(int steps) {
+    if (!m_pLibraryWidget || steps == 0) {
         return;
     }
 
-    if (v > 0) {
-        LibraryView* pActiveView = m_pLibraryWidget->getActiveView();
-        if (!pActiveView) {
-            return;
-        }
-        pActiveView->assignPreviousTrackColor();
-    }
-}
-
-void LibraryControl::slotTrackColorNext(double v) {
-    if (!m_pLibraryWidget) {
+    LibraryView* pActiveView = m_pLibraryWidget->getActiveView();
+    if (!pActiveView) {
         return;
     }
 
-    if (v > 0) {
-        LibraryView* pActiveView = m_pLibraryWidget->getActiveView();
-        if (!pActiveView) {
-            return;
+    while (steps != 0) {
+        if (steps > 0) {
+            pActiveView->assignNextTrackColor();
+            steps--;
+        } else {
+            pActiveView->assignPreviousTrackColor();
+            steps++;
         }
```

Comment:
Is this possible without the while? It's not very efficient and could cause visual artifacts.

Reply:
Thanks for your review.
colorPalette.nextColor() only handles color codes, the actual QT event is thrown with track->setColor().
So I think no visual artifacts or performance issue can be caused by this.
Another code using this have been push last week (see https://github.com/mixxxdj/mixxx/pull/13066)

Reply:
I think @Swiftb0y is referring to the Track::colorUpdated(color) signal that is fired for each color change whic causes repaint calls everywhere that track (or many tracks) is loaded (tracks view, decks), and with the `selector` color changes can indeed occur at high rates.
Didn't think of that during my previous review :+1:

Reply:
So ideally the color selector would be moved to the track itself, find the desired color and assign it right away.
IIUC that only refers to the situation when we have select steps buffered, e.g. from MIDI, and LibraryControl is going to do one by one. Because 'quick' color switch may also happen when triggering color_next/_prev quickly.

Reply:
> causes repaint calls everywhere that track (or many tracks)

okay, `WTrackTableView::assignNextTrackColor` acts only on the first selected track.

Reply:
> causes repaint calls everywhere

Yes. thats what I was referring to. 

> IIUC that only refers to the situation when we have select steps buffered,

That seems like an ugly workaround. Ideally the API would just be changed so you can just set n'th next color directly. 

> from MIDI, and LibraryControl is going to do one by one.

If the user quickly cycles through colors by pressing the next/prev button quickly, thats fine. Especially since there is quite a lot of delay between the individual presses (which is not true for a programmed loop).

Reply:
Okay, I was looking at the wrong patch when I first replied (the one in basetrackplayer)
So I updated this one following your advice, what do you think ?